test/fw/ptl/utils/plugins/ptl_test_data.py

```diff
@@ -55,240 +56,260 @@ class PTLTestData(Plugin):
+    def set_data(self, post_data_dir, max_postdata_threshold):
+        self.post_data_dir = post_data_dir
+        self.max_postdata_threshold = max_postdata_threshold
 
     def configure(self, options, config):
         """
         Configure the plugin and system, based on selected options
         """
         self.config = config
-        self.enabled = True
+        if self.post_data_dir is not None:
+            self.enabled = True
+        else:
+            self.enabled = False
 
-    def __get_sntnbi_name(self, test):
+    def __save_home(self, test, status, err=None):
         if hasattr(test, 'test'):
             _test = test.test
             sn = _test.__class__.__name__
         elif hasattr(test, 'context'):
             _test = test.context
             sn = _test.__name__
         else:
-            return ('unknown', 'unknown', 'unknown')
-        tn = getattr(_test, '_testMethodName', 'unknown')
-        if (hasattr(_test, 'server') and
-                (getattr(_test, 'server', None) is not None)):
-            bi = _test.server.attributes['pbs_version']
-        else:
-            bi = 'unknown'
-        return (sn, tn, bi)
-
-    def __save_home(self, test, status):
-        if hasattr(test, 'test'):
-            _test = test.test
-        elif hasattr(test, 'context'):
-            _test = test.context
-        else:
             # test does not have any PBS Objects, so just return
             return
-        if not hasattr(_test, 'server'):
-            # test does not have any PBS Objects, so just return
+        if self.__priv_sn != sn:
+            self.__save_data_count = 0
+            self.__priv_sn = sn
+        # Saving home might take time so disable timeout
+        # handler set by runner
+        tn = getattr(_test, '_testMethodName', 'unknown')
+        testlogs = getattr(test, 'captured_logs', '')
+        datadir = os.path.join(self.post_data_dir, sn, tn)
+        if os.path.exists(datadir):
+            _msg = 'Old post analysis data exists at %s' % datadir
+            _msg += ', skipping saving data for this test case'
+            self.logger.warn(_msg)
+            _msg = 'Please remove old directory or'
+            _msg += ' provide different directory'
+            self.logger.warn(_msg)
             return
-        st = getattr(test, 'start_time', None)
-        if st is not None:
-            st = time.mktime(st.timetuple())
+        if getattr(test, 'old_sigalrm_handler', None) is not None:
+            _h = getattr(test, 'old_sigalrm_handler')
+            signal.signal(signal.SIGALRM, _h)
+            signal.alarm(0)
+        self.logger.log(logging.DEBUG2, 'Saving post analysis data...')
+        current_host = socket.gethostname().split('.')[0]
+        self.du.mkdir(current_host, path=datadir, mode=0755,
+                      parents=True, logerr=False, level=logging.DEBUG2)
+        if err is not None:
+            if isclass(err[0]) and issubclass(err[0], SkipTest):
```

Comment:
Don't we call __save_home when there is a test case failure? Should we be checking it against SkipTest?  OR is it that we call __save_home for every test case which would mean that it will also gets called for tests that were skipped or failed or even passed? If so, why do we do that? wouldn't it cause delay?
AND
Why do we check for err here when we have already created a directory? shouldn't we be moving it up in this function?